flocker/acceptance/endtoend/test_installer.py

```diff
@@ -0,0 +1,319 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for AWS CloudFormation installer.
+"""
+
+from datetime import timedelta
+import json
+import os
+from subprocess import check_call, check_output
+import time
+
+from twisted.internet import reactor
+from twisted.internet.task import deferLater
+from twisted.internet.defer import maybeDeferred
+
+from twisted.python.filepath import FilePath
+from eliot import Message
+
+
+from ...common.runner import run_ssh
+from ...common import gather_deferreds
+from ...testtools import AsyncTestCase, async_runner
+from ..testtools import Cluster, ControlService
+from ...ca import treq_with_authentication, UserCredential
+from ...apiclient import FlockerClient
+from ...control.httpapi import REST_API_PORT
+
+REGION = os.environ['REGION']
+ACCESS_KEY_ID = os.environ['ACCESS_KEY_ID']
+SECRET_ACCESS_KEY = os.environ['SECRET_ACCESS_KEY']
+
+PARAMETERS = [
+    {
+        'ParameterKey': 'KeyName',
+        'ParameterValue': os.environ['KEY_NAME']
+    },
+    {
+        'ParameterKey': 'AccessKeyID',
+        'ParameterValue': os.environ['ACCESS_KEY_ID']
+    },
+    {
+        'ParameterKey': 'SecretAccessKey',
+        'ParameterValue': os.environ['SECRET_ACCESS_KEY']
+    }
+]
+
+COMPOSE_NODE0 = '/home/ubuntu/postgres/docker-compose-node0.yml'
+COMPOSE_NODE1 = '/home/ubuntu/postgres/docker-compose-node1.yml'
+RECREATE_STATEMENT = 'drop table if exists test; create table test(i int);'
+INSERT_STATEMENT = 'insert into test values(1);'
+SELECT_STATEMENT = 'select count(*) from test;'
+
+CLOUDFORMATION_STACK_NAME = 'testinstallerstack'
+S3_CLOUDFORMATION_TEMPLATE = (
+    'https://s3.amazonaws.com/'
+    'installer.downloads.clusterhq.com/flocker-cluster.cloudformation.json'
+)
+
+
+def remote_command(node, *args):
+    command_output = []
+    d = run_ssh(
+        reactor,
+        'ubuntu',
+        node,
+        args,
+        handle_stdout=command_output.append
+    )
+    d.addCallback(
+        lambda process_result: (process_result, command_output)
+    )
+    return d
+
+
+def remote_docker_compose(client_ip, docker_host, compose_file_path, *args):
+    docker_compose_output = []
+    d = run_ssh(
+        reactor,
+        'ubuntu',
+        client_ip,
+        ('DOCKER_HOST={}'.format(docker_host),
+         'docker-compose', '--file', compose_file_path) + args,
+        handle_stdout=docker_compose_output.append
+    )
+    d.addCallback(
+        lambda process_result: (process_result, docker_compose_output)
+    )
+    return d
+
+
+def remote_postgres(client_ip, host, command):
+    postgres_output = []
+    d = deferLater(
+        reactor,
+        60,
```

Comment:
Why 60? Maybe add a comment or define a constant or both. 

If it's because something else needs to happen first, then instead of waiting a fixed length of time, we should only try after that condition is fixed (otherwise we're just adding more flaky tests). `flocker.common.retry` has a bunch of helpers for this.